apps/api/v2/src/ee/bookings/2024-08-13/services/output.service.ts

```diff
@@ -0,0 +1,128 @@
+import { BookingsRepository_2024_08_13 } from "@/ee/bookings/2024-08-13/bookings.repository";
+import { Injectable } from "@nestjs/common";
+import { plainToClass } from "class-transformer";
+import { DateTime } from "luxon";
+import { z } from "zod";
+
+import { BookingOutput_2024_08_13, RecurringBookingOutput_2024_08_13 } from "@calcom/platform-types";
+import { Booking } from "@calcom/prisma/client";
+
+export const bookingResponsesSchema = z.object({
+  email: z.string(),
+  name: z.string(),
+  guests: z.array(z.string()).optional(),
+  rescheduledReason: z.string().optional(),
+});
+
+type DatabaseBooking = Booking & {
+  attendees: {
+    name: string;
+    email: string;
+    timeZone: string;
+    locale: string | null;
+    noShow: boolean | null;
+  }[];
+} & { user: { id: number; name: string | null; email: string } | null };
+
+@Injectable()
+export class OutputBookingsService_2024_08_13 {
+  constructor(private readonly bookingsRepository: BookingsRepository_2024_08_13) {}
+
+  getOutputBooking(databaseBooking: DatabaseBooking) {
+    const dateStart = DateTime.fromISO(databaseBooking.startTime.toISOString());
+    const dateEnd = DateTime.fromISO(databaseBooking.endTime.toISOString());
+    const duration = dateEnd.diff(dateStart, "minutes").minutes;
+
+    const bookingResponses = bookingResponsesSchema.parse(databaseBooking.responses);
+    const attendee = databaseBooking.attendees.find((attendee) => attendee.email === bookingResponses.email);
+
+    if (!attendee) {
+      throw new Error("Attendee not found");
+    }
+
+    const booking = {
+      id: databaseBooking.id,
+      uid: databaseBooking.uid,
+      hosts: [databaseBooking.user],
+      status: databaseBooking.status.toLowerCase(),
+      cancellationReason: databaseBooking.cancellationReason || undefined,
+      reschedulingReason: bookingResponses?.rescheduledReason,
+      rescheduledFromUid: databaseBooking.fromReschedule || undefined,
+      start: databaseBooking.startTime,
+      end: databaseBooking.endTime,
+      duration,
+      eventTypeId: databaseBooking.eventTypeId,
+      attendees: databaseBooking.attendees.map((attendee) => ({
+        name: attendee.name,
+        email: attendee.email,
+        timeZone: attendee.timeZone,
+        language: attendee.locale,
+        absent: !!attendee.noShow,
+      })),
+      guests: bookingResponses.guests,
+      meetingUrl: databaseBooking.location,
+      absentHost: !!databaseBooking.noShowHost,
+    };
+
+    return plainToClass(BookingOutput_2024_08_13, booking, { strategy: "excludeAll" });
+  }
+
+  async getOutputRecurringBookings(databaseBookings: DatabaseBooking[]) {
+    const transformed = [];
+
+    for (const booking of databaseBookings) {
+      if (!booking.id) {
+        throw new Error("Booking was not created");
+      }
```

Comment:
can we end up having created bookings but only return an error because one was not ?

Reply:
Check was left from previous code - removed in [refactor: remove unused check](https://github.com/calcom/cal.com/pull/16200/commits/f9944a06f1024d45c091b78b7c90a89a628f05e0)